pennylane/io/qualtran_io.py

```diff
@@ -0,0 +1,338 @@
+            from qualtran.bloqs.chemistry.trotter.ising import IsingXUnitary, IsingZZUnitary
+            from qualtran.bloqs.chemistry.trotter.trotterized_unitary import TrotterizedUnitary
+
+            # Parameters for the TrotterizedUnitary
+            nsites = 5
+            j_zz, gamma_x = 2, 0.1
+            zz_bloq = IsingZZUnitary(nsites=nsites, angle=0.02 * j_zz)
+            x_bloq = IsingXUnitary(nsites=nsites, angle=0.01 * gamma_x)
+            trott_unitary = TrotterizedUnitary(
+                bloqs=(x_bloq, zz_bloq),  timestep=0.01,
+                indices=(0, 1, 0), coeffs=(0.5 * gamma_x, j_zz, 0.5 * gamma_x)
+            )
+
+            # Instantiate the TextbookQPE and pass in our unitary
+            textbook_qpe = TextbookQPE(trott_unitary, RectangularWindowState(3))
+
+            # Execute on device
+            dev = qml.device("default.qubit")
+            @qml.qnode(dev)
+            def circuit():
+                qml.FromBloq(textbook_qpe, wires=range(textbook_qpe.signature.n_qubits()))
+                return qml.probs(wires=[5, 6, 7])
+
+        >>> circuit()
+        array([0.94855734, 0.01291602, 0.00431044, 0.00267219, 0.00237645,
+        0.00290337, 0.00526717, 0.02099701])
+
+    .. details::
+        :title: Usage Details
+
+        The decomposition of a bloq wrapped in ``qml.FromBloq`` may use more wires than expected.
+        For example, when we wrap Qualtran's ``CZPowGate``, we get
+
+        >>> from qualtran.bloqs.basic_gates import CZPowGate
+        >>> qml.FromBloq(CZPowGate(0.468, eps=1e-11), wires=[0, 1]).decomposition()
+        [FromBloq(And, wires=Wires([0, 1, 'alloc_free2'])),
+        FromBloq(Z**0.468, wires=Wires(['alloc_free2'])),
+        FromBloq(And†, wires=Wires([0, 1, 'alloc_free2']))]
+
+        This behaviour results from the decomposition of ``CZPowGate`` as defined in Qualtran,
+        which allocates and frees a wire all in the same ``bloq``. In this situation,
+        PennyLane automatically allocates this wire under the hood, and that additional wire is
+        named ``alloc_free_{idx}``. The indexing starts at the length of the wires defined in the
+        signature, which in the case of ``CZPowGate`` is :math:`2`. Due to the current
+        limitations of PennyLane, these wires cannot be accessed manually or mapped.
+
+    """
+
+    def __init__(self, bloq, wires: WiresLike):
+        if not isinstance(bloq, qt.Bloq):
+            raise TypeError(f"bloq must be an instance of {qt.Bloq}.")
+        self._hyperparameters = {"bloq": bloq}
+        super().__init__(wires=wires, id=None)
+
+    def __repr__(self):
+        return f'FromBloq({self.hyperparameters["bloq"]}, wires={self.wires})'
+
+    @staticmethod
+    def compute_decomposition(
+        *params,
+        wires=None,
+        **hyperparameters,
+    ):  # pylint: disable=arguments-differ, unused-argument, too-many-branches, too-many-statements
```

Comment:
I don't see where the data is getting set fo this op. What's the data?
But potentially, we can just do:
```suggestion
    def compute_decomposition(
        wires,
        bloq
    ):  # pylint: disable=arguments-differ, unused-argument, too-many-branches, too-many-statements
```

Reply:
I did a bit of minor refactoring but the other parts aren't easily pulled out of the function body.